File: Week 6/sentimental-mario-more/mario.py
# TODO
import cs50

def make(blanks, blocks):
    if blanks == 0:
        return

    blanks -= 1
    blocks += 1

    print(" " * blanks, end = "")
    print("#" * blocks, end = "  ")
    print("#" * blocks)
    make(blanks, blocks)


# Ask for height of pyramid
while True:
    height = cs50.get_int("Input height of pyramid: ")
    if height >= 1 and height <= 8:
        break

make(height, 0)

File: Pset 1/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int height;
    int width = 1;

    // ask for height repeatedly
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    // traverse rows
    for (int i = 0; i < height; i++)
    {
        // traverse rows
        for (int j = 0; j < height - width; j++)
        {
            printf(" ");
        }

        // hashes
        for (int k = 0; k < width; k++)
        {
            printf("#");
        }

        printf("  ");

        for (int l = 0; l < width; l++)
        {
            printf("#");
        }
        
        // increment counter for num hashes
        width++;
        printf("\n");
    }

}